### ISSUE TITLE
No heartbeat from dserver/hdb++es-srv/arch

### ISSUE THREAD

**Issue by prokofyevDmitry** [tango-controls-hdbpp/hdbpp-configurator#1]
Hello, i'm in the process of installation of hdbpp suite and while the system works (i can configure and visualize my reccords) I encountered a bug/unusual behaviour as after few seconds the list of attributes to archive the program prints the message **No heartbeat from dserver/hdb++es-srv/arch**.
![image](https://user-images.githubusercontent.com/8905771/37095485-6534b228-2216-11e8-9bd2-67b11dd3e2da.png)

## Expected Behavior
On start-up the program should list all attributes that were added to a subscriber device server. 
![image](https://user-images.githubusercontent.com/8905771/37095696-eb8ccee6-2216-11e8-8f15-0289e6b0d0f2.png)

## Current Behavior
On start-up the program list all attributes during few seconds then prints the message **No heartbeat from dserver/hdb++es-srv/arch** and the program become unusable because if we try to delete this message as we would do with an attribute a **division by 0** error occurs. 

## Possible Solution
During the few seconds affter the start up I was able to configure and launch the archiving of few attributes from the Tango Test device server and I was able to visualize those records in the Viewer tool. After reading [this](http://www.tango-controls.org/community/forum/c/general/development/no-heartbeat-error-on-event-subscription/?page=1#post-616) forum thread it seems that even if a lost of heartbeat from a device server occurs the events continue to be triggered so one possibility might be the simple suppression of the warning or better a popup that alert us of lost of heartbeat instead of blocking the entire program.

**Comment by Pascal-Verdier** [comment]
Hi
I know about this problem. And I have to fix it.
But normally it append only if you stop and re start your DS subscriber.
Is it your case ?
Or do you have a real problem to receive heartbeats  ?

**Comment by prokofyevDmitry** [comment]
Hi thx for the fast response :) 

This happens every time without restarting the DS subscriber. 
What is strange is that on start up the DS is well detected and i have some 10 sec during which I can use the program normally and only then the hearbeat is lost. 

If you kindly could point me to the file where you fill the list with attributes (and in this case the error) I can thinker on my end and then make a pull req.

**Comment by Pascal-Verdier** [comment]
This GUI subscribes to events on /*hdb++cm-srv*/  and /*hdb++es-srv*/ 
devices.
As all TANGO event managements, if a heartbeat is not receive every 10 
seconds the system is faulty.
That means there is a network configuration problem between your devices 
and the GUI.
Difficult to have a diagnostic from here.
Could be a problem in address resolving.

For information which JTango release are you using ?
Could you send me a screen dump with Jive with a /*hdb++es-srv*/ devices 
selected ?

**Comment by prokofyevDmitry** [comment]
I played a little bit with the network settings and I managed to make it work. 

Before my Tango Host was  localhost (127.0.0.1). 
I edited the Tango Host and the hdb++es-srv settings in order to use my machine's ip as Tango Host and it worked. 
I cannot clearly comprehend why using the local loop is problematic for event subscription, I will look into it later. 
Thanks for the hint about the network configuration issue. 
I guess I can close this issue. 
Good Day :)
